keras/src/saving/saving_lib.py

```diff
@@ -432,10 +444,17 @@ def _load_model_from_fileobj(fileobj, custom_objects, compile, safe_mode):
         asset_store = None
         try:
             if _VARS_FNAME_H5 in all_filenames:
-                if isinstance(fileobj, io.BufferedReader):
-                    # First, extract the model.weights.h5 file, then load it
-                    # using h5py.
-                    try:
+                try:
+                    if not is_large_model(model):
+                        # Load the entire file into memory if the model is not
+                        # large.
+                        io_file = io.BytesIO(
+                            zf.open(_VARS_FNAME_H5, "r").read()
+                        )
+                        weights_store = H5IOStore(io_file, mode="r")
+                    else:
+                        # Try extracting the model.weights.h5 file, and then
+                        # loading it using using h5py.
```

Comment:
Add: "This is significantly faster than reading from the zip archive on the fly" (for future maintenance)

Reply:
Added.